### PR TITLE
get addresses for job iteration object

### PR DESCRIPTION
**Pull request by venkataanil**
RFE https://github.com/kube-burner/kube-burner-ocp/issues/32

GetIPAddress renders a string with IP addresses for the object template from the list of input addresses based on iteration number and number of addresses requested per object

### CODE REVIEW

**Comment by vishnuchalla** [comment]
Looks good to me. Requires signed commit I guess.

File: pkg/util/template.go [hunk 1]
```diff
@@ -43,6 +43,17 @@ func RenderTemplate(original []byte, inputData interface{}, options templateOpti
 	funcMap["GetSubnet24"] = func(subnetIdx int) string {
 		return netip.AddrFrom4([4]byte{byte(subnetIdx>>16 + 1), byte(subnetIdx >> 8), byte(subnetIdx), 0}).String() + "/24"
 	}
+	funcMap["GetIPAddress"] = func(eipAddresses string, Iteration int, addressesPerIteration int) string {
+		fmt.Println("Iteration %s addressesPerIteration %s", Iteration, addressesPerIteration)
```
Comment (rsevilla87) [inline pkg/util/template.go:47]: Don't use Println

File: pkg/util/template.go [hunk 2]
```diff
@@ -43,6 +43,17 @@ func RenderTemplate(original []byte, inputData interface{}, options templateOpti
 	funcMap["GetSubnet24"] = func(subnetIdx int) string {
 		return netip.AddrFrom4([4]byte{byte(subnetIdx>>16 + 1), byte(subnetIdx >> 8), byte(subnetIdx), 0}).String() + "/24"
 	}
+	funcMap["GetIPAddress"] = func(eipAddresses string, Iteration int, addressesPerIteration int) string {
```
Comment (rsevilla87) [inline pkg/util/template.go:46]: What are you trying to achieve with this function?

Comment (venkataanil) [reply pkg/util/template.go:46]: I have created an RFE for this https://github.com/kube-burner/kube-burner-ocp/issues/32

egress ip object, which is part of job need list of ip addresses. For example 10.0.0.2 and 10.0.0.3 in the below case.

> apiVersion: k8s.ovn.org/v1
kind: EgressIP
metadata:
name: egressip-obj-0-1
spec:
egressIPs:
- 10.0.0.2
- 10.0.0.3

GetIPAddress() function is called from the template while rendering. Inputs are -
**eipAddresses:**  all avaialable IP addresses kube-burner-ocp has provided . Example "10.0.0.2 10.0.0.3 10.0.0.4 10.0.0.5 10.0.0.6 10.0.0.7 10.0.0.8 10.0.0.9 10.0.0.10"
**Iteration:** Iteration cout, example 0
**addrPerIteration:** Number of addresses needed by object, for example 2

> apiVersion: k8s.ovn.org/v1
kind: EgressIP
metadata:
  name: egressip-obj-{{.Iteration}}-{{.Replica}}
spec:
  egressIPs:
  {{range (splitList " " (GetIPAddress .eipAddresses .Iteration .addrPerIteration))}}
  - {{.}}
  {{end}}

Similarly deployment-client.yml also calls this function to get same IP addresses what the eip object got, for verfication

> apiVersion: v1
kind: Pod
spec:
containers:
…
image: quay.io/cloud-bulldozer/curl:latest
readinessProbe:
exec:
command:
- /bin/sh
- -c
- curl -s http://${SERVER_HOST}:${SERVER_PORT} | grep -qE '10.0.0.2|10.0.0.3'


Template -
https://github.com/kube-burner/kube-burner-ocp/pull/28/files#diff-8343c793121c8537ec152195e9befb0587f431ef6be2d2be3c89c18845981e0c 

> kind: Deployment
apiVersion: apps/v1
metadata:
  name: client-{{.Replica}}
spec:
spec:
      containers:
      - name: client-app
        image: quay.io/cloud-bulldozer/curl:latest
        command: ["sleep", "inf"]
        resources:
          requests:
            memory: "10Mi"
            cpu: "10m"
        env:
            - "/bin/sh"
            - "-c"
            {{- $eips := (splitList " " (GetIPAddress .eipAddresses .Iteration .addrPerIteration) | join "|") }}
            - "curl -s http://${SERVER_HOST}:${SERVER_PORT} | grep -qE '{{$eips}}'"

File: pkg/util/template.go [hunk 3]
```diff
@@ -43,6 +43,17 @@ func RenderTemplate(original []byte, inputData interface{}, options templateOpti
...
+		fmt.Println("Iteration %s addressesPerIteration %s", Iteration, addressesPerIteration)
+	        iteration := Iteration
```
Comment (rsevilla87) [inline pkg/util/template.go:48]: remove this line

Comment (vishnuchalla) [reply pkg/util/template.go:48]: @venkataanil as raul suggested, can you remove that line as it seems not required.

**Review by vishnuchalla — CHANGES_REQUESTED** [review]
.

File: pkg/util/template.go (hunk 3 above)
Comment (venkataanil) [reply pkg/util/template.go:48]: sure

**Review by vishnuchalla — APPROVED** [review]
lgtm

**Pull request merged**